ground/gcs/src/plugins/uavobjects/uavobjectfield.cpp

```diff
@@ -877,12 +882,14 @@ QVariant UAVObjectField::getValue(quint32 index)
     {
         quint8 tmpenum;
         memcpy(&tmpenum, &data[offset + numBytesPerElement*index], numBytesPerElement);
-        //            Q_ASSERT((tmpenum < options.length()) && (tmpenum >= 0)); // catch bad enum settings
-        if(tmpenum >= options.length()) {
-            qDebug() << "Invalid value for" << name;
-            return QVariant( QString("Bad Value") );
-        }
-        return QVariant( options[tmpenum] );
+	// Too slow?
+	for (int i = 0; i < indices.length(); i++) {

```

Comment:
Basically because there's existing stuff that relies on the options being a list-- to generate combo boxes, export things, etc.  I didn't want to break it.

Anyways, the other direction this data structure is fast.  It just requires a scan the other way.  Could add a hash to speed this way up but I've not seen any performance penalty on save.